### PR TITLE
adding assert messages to std.container.binary_heap

### PR DESCRIPTION
**Pull request by burner**

### CODE REVIEW

File: std/container/binaryheap.d [hunk 1]
```diff
@@ -386,7 +388,8 @@ leaves the heap unaffected and returns `false`.
     bool conditionalSwap(ref ElementType!Store value)
     {
         _payload.refCountedStore.ensureInitialized();
-        assert(_length == _store.length);
+        assert(_length == _store.length,
+                "lenght and number of stored items out of sync");
```
Comment (berni44) [inline std/container/binaryheap.d]: lenght => length

Comment (burner) [reply std/container/binaryheap.d]: thank you

Comment (burner) [reply std/container/binaryheap.d]: thank you

**Comment by burner** [comment]
thank you @berni44 @n8sh

**Pull request merged**